crates/e2e/tests/e2e/pre_interaction.rs

```diff
@@ -101,4 +114,26 @@ async fn pre_interaction(web3: Web3) {
     tracing::info!("Waiting for auction to be cleared.");
     let auction_is_empty = || async { services.get_auction().await.auction.orders.is_empty() };
     wait_for_condition(TIMEOUT, auction_is_empty).await.unwrap();
+
+    // Check malicious custom interactions did not work.
+    let allowance = cow
+        .allowance(
+            onchain.contracts().gp_settlement.address(),
+            trader.address(),
+        )
+        .call()
+        .await
+        .unwrap();
+    assert_eq!(allowance, U256::zero());
```

Comment:
This seems to suggest that custom interactions can revert without reverting the whole transaction.
I guess this is because the multisend trampoline call calls these interactions with a `try{} catch(e){}`?
Would be nice to document that somewhere.

Reply:
> This seems to suggest that custom interactions can revert without reverting the whole transaction.

No, this is not the case. It is just the allowance interaction is not made from the settlement contract index.

I will add something like this to document why this works:

```rust
    let allowance = cow
        .allowance(
            onchain.contracts().multisend.address(),
            trader.address(),
        )
        .call()
        .await
        .unwrap();
    assert_eq!(allowance, U256::max_value());
```